OPHD/States/MapViewState.cpp

```diff
@@ -69,6 +69,43 @@ const std::map<Robot::Type, RobotMeta> RobotMetaTable
 };
 
 
+static NAS2D::Rectangle<int> buildAreaRectFromTile(const Tile& centerTile, int commRange)
+{
+	const NAS2D::Point commAreaStartPoint
+	{
+		std::clamp(centerTile.position().x - commRange, 0, 299),
+		std::clamp(centerTile.position().y - commRange, 0, 149)
+	};
+
+	const NAS2D::Point commAreaEndPoint
+	{
+		std::clamp(centerTile.position().x + commRange, 0, 299),
+		std::clamp(centerTile.position().y + commRange, 0, 149)
+	};
```

Comment:
Magic numbers here. I assume those constants relate to standard map dimensions. We should be using a named constant, especially since the values are repeated.

----

We should probably add some kind of `Rectangle::clamp` method, to allow clamping a computed `Rectangle` within some knows bounds, such as map bounds.

----

Actually, due to the symmetry from the center point, you might be able to use `Rectangle::inset` for this, with a negative inset amount:
```cpp
const auto commArea = Rectangle<int>::Create(centerTile.position(), {0, 0}).inset(-commRange);
```

Of course that that would require clamping.

Reply:
> Magic numbers here. I assume those constants relate to standard map dimensions. We should be using a named constant, especially since the values are repeated.

Agreed. This was a bit of a quick-and-dirty type of deal. I was kind of surprised I didn't have named constants or that `TileMap` didn't have an option. Upon further inspection I see that `TileMap` does have `TileMap::size()` which would have served for this but for whatever reason I didn't think of looking for a `size()` type of function. For maps I usually think of a `width` and `height` function that gets me exactly the values I need.

Definitely something to address in a future change set.

> 
> We should probably add some kind of `Rectangle::clamp` method, to allow clamping a computed `Rectangle` within some knows bounds, such as map bounds.
> 
> Actually, due to the symmetry from the center point, you might be able to use `Rectangle::inset` for this, with a negative inset amount:
> 
> ```c++
> const auto commArea = Rectangle<int>::Create(centerTile.position(), {0, 0}).inset(-commRange);
> ```
> 
> Of course that that would require clamping.

We could? But this seems to work just fine here?